src/plot.js

```diff
@@ -96,12 +95,38 @@ export function plot(options = {}) {
       .call(applyInlineStyles, style)
     .node();
 
-  for (const mark of marks) {
+  // A mark layout is responsible for:
+  // * applying the scales, possibly pushing a new one (for legends)
+  // * returning a filtered index and values to render, possibly new channels to rescale
+  const newScaleChannels = new Map();
+  const layouts = [];
+  const newOptions = {};
+  for (let i = 0; i < marks.length; ++i) {
+    const mark = marks[i];
+    const {channels} = layouts[i] = applyLayout.call(mark, mark.layout, [markIndex.get(mark)], scaleDescriptors, markChannels.get(mark), dimensions, options);
+    if (channels) {
+      for (const [, {scale, options: hint}] of channels) {
+        if (scale) newOptions[scale] = {...hint, ...options[scale]};
```

Comment:
If you have multiple layouts that define multiple channels for a given scale, or even a single layout that defines multiple channels for the same scale, this won’t compose the scale options; instead, it’s “last one wins”, i.e., the last channel definition for a given scale is respected and the other options are ignored. Probably we need to accumulate into newOptions. But I’m also curious if we can sidestep this approach entirely by using channel hints rather than letting layouts inject scale options directly.